release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md

```diff
@@ -1,22 +1,142 @@
+
+```rust
+// Before
+let color = Color::rgb(1.0, 0.0, 1.0).as_hsla();
+
+// After
+let color: Hsla = Srgba::rgb(1.0, 0.0, 1.0).into();
+```
+
+#### `Color` methods
+
+Any mention of RGB has been renamed to [sRGB]. This includes the variant `Color::Rgba` turning into `Color::Srgba` as well as methods such as `Color::rgb` and `Color::rgb_u8` turning into `Color::srgb` and `Color::srgb_u8`.
+
+[sRGB]: https://en.wikipedia.org/wiki/SRGB
+
+Methods to access specific channels of `Color` have been removed due to causing silent, relatively expensive conversions. This includes `Color::r`, `Color::set_r`, `Color::with_r`, and all of the equivalents for `g`, `b` `h`, `s` and `l`. Convert your `Color` into the desired color space, perform your operation there, and then convert it back.
+
+```rust
+// Before
+let mut color = Color::rgb(0.0, 0.0, 0.0);
+color.set_b(1.0);
+
+// After
+let color = Color::srgb(0.0, 0.0, 0.0);
+let srgba = Srgba {
+    blue: 1.0,
+    ..Srgba::from(color),
+};
+let color = Color::from(srgba);
+```
+
+`Color::hex` has been moved to `Srgba::hex`. Call `.into()` or construct a `Color::Srgba` variant manually to convert it.
+
+`Color::rgb_linear` and `Color::rgba_linear` have been renamed `Color::linear_rgb` and `Color::linear_rgba` to fit the naming scheme of the `LinearRgba` struct.
+
+`Color::as_linear_rgba_f32` and `Color::as_linear_rgba_u32` have been removed. Call `LinearRgba::to_f32_array` and `LinearRgba::to_u32` instead, converting if necessary.
+
+Several other color conversion methods to transform LCH or HSL colors into float arrays or `Vec` types have been removed. Please reimplement these externally or open a PR to re-add them if you found them particularly useful.
+
+Vector field arithmetic operations on `Color` (add, subtract, multiply and divide by a f32) have been removed. Instead, convert your colors into `LinearRgba` space and perform your operations explicitly there. This is particularly relevant when working with emissive or HDR colors, whose color channel values are routinely outside of the ordinary 0 to 1 range.
+
+#### Alpha
+
+Alpha, also known as transparency, used to be referred to by the letter `a`. It is now called by its full name within structs and methods.
+
+- `Color::set_a` and `Color::a` are now `Color::set_alpha` and `Color::alpha`. These are part of the new `Alpha` trait.
+- Additionally, `Color::is_fully_transparent` is now part of the `Alpha`.
+
+#### CSS Constants
+
+The various CSS color constants are no longer stored directly on `Color`. Instead, they’re defined in the `Srgba` color space, and accessed via `bevy::color::palettes::css`. Call `.into()` on them to convert them into a `Color` for quick debugging use, and consider using the much prettier `tailwind` palette for prototyping. Furthermore, the `LIME_GREEN` color has been renamed to `LIMEGREEN` to comply with the standard naming.
```

Comment:
is it worth also mentioning the basic palette? I'd also specifically call out that basic::GREEN is not what Color::GREEN was? (this caught me out migrating some code)

Reply:
Probably! I'll at least add a mention that, with the new palettes, the old colors may not be copied 1:1.

Reply:
I believe I addressed this in ab89f31bbf8072b3a445aa949b42ae6fe53325bf. If you have a different suggestion, feel free to re-open this thread.